### PR TITLE
feat(mayactl) Enhance run task to get controller host name and fix NS for cstor response

### PR DESCRIPTION
**Pull request by ashishranjan738**
Signed-off-by: Ashish Ranjan <ashishranjan738@gmail.com>

**What this PR does / why we need it**:
-> Enhanced volume read runtask to give controller node name which can be consumed by mayactl .
-> Fixes empty namespace for cstor volumes in volume list.

**Special notes for your reviewer**:
-> volume list response: 
```
{
  "items": [
    {
      "apiVersion": "v1alpha1",
      "kind": "CASVolume",
      "metadata": {
        "annotations": {
          "openebs.io/replica-ips": "10.4.1.8, 10.4.0.10, 10.4.2.8",
          "openebs.io/replica-status": "running, running, running",
          "openebs.io/volume-size": "5G",
          "vsm.openebs.io/cluster-ips": "10.7.254.120",
          "vsm.openebs.io/replica-count": "3",
          "vsm.openebs.io/targetportals": "10.7.254.120:3260",
          "openebs.io/iqn": "iqn.2016-09.com.openebs.jiva:default-jiva-testvol-1-1931265930",
          "vsm.openebs.io/controller-status": "running,running",
          "vsm.openebs.io/volume-size": "5G",
          "openebs.io/cluster-ips": "10.7.254.120",
          "openebs.io/targetportals": "10.7.254.120:3260",
          "vsm.openebs.io/iqn": "iqn.2016-09.com.openebs.jiva:default-jiva-testvol-1-1931265930",
          "vsm.openebs.io/replica-ips": "10.4.1.8, 10.4.0.10, 10.4.2.8",
          "openebs.io/controller-ips": "10.4.0.9",
          "openebs.io/replica-count": "3",
          "vsm.openebs.io/controller-ips": "10.4.0.9",
          "vsm.openebs.io/replica-status": "running, running, running",
          "openebs.io/controller-status": "running,running"
        },
        "name": "default-jiva-testvol-1-1931265930",
        "namespace": "default"
      },
      "spec": {
        "capacity": "5G",
        "casType": "jiva",
        "fsType": "",
        "iqn": "iqn.2016-09.com.openebs.jiva:default-jiva-testvol-1-1931265930",
        "lun": 0,
        "replicas": "1",
        "targetIP": "10.7.254.120",
        "targetPort": "3260",
        "targetPortal": "10.7.254.120:3260"
      },
      "status": {
        "Message": "",
        "Phase": "",
        "Reason": ""
      }
    },
    {
      "apiVersion": "v1alpha1",
      "kind": "CASVolume",
      "metadata": {
        "annotations": {
          "openebs.io/cluster-ips": "10.7.248.146",
          "openebs.io/controller-status": "running running running",
          "openebs.io/volume-size": "5G"
        },
        "name": "default-cstor-testvol-1-678850436",
        "namespace": "openebs"
      },
      "spec": {
        "capacity": "5G",
        "casType": "cstor",
        "fsType": "",
        "iqn": "iqn.2016-09.com.openebs.cstor:default-cstor-testvol-1-678850436",
        "lun": 0,
        "replicas": "3",
        "targetIP": "10.7.248.146",
        "targetPort": "3260",
        "targetPortal": "10.7.248.146:3260"
      },
      "status": {
        "Message": "",
        "Phase": "",
        "Reason": ""
      }
    }
  ],
  "metadata": {},
  "metalist": {}
}
```
-> Volume read output for cstor:
```
{
  "apiVersion": "v1alpha1",
  "kind": "CASVolume",
  "metadata": {
    "annotations": {
      "openebs.io/pool-names": "cstor-sparse-pool-2bas,cstor-sparse-pool-5bvb,cstor-sparse-pool-tf4k",
      "openebs.io/controller-ips": "10.4.2.13",
      "openebs.io/controller-node-name": "gke-ashish-dev-default-pool-c5bc7514-j4gj",
      "openebs.io/controller-status": "running,running,running",
      "openebs.io/cvr-names": "default-cstor-testvol-1-678850436-cstor-sparse-pool-2bas,default-cstor-testvol-1-678850436-cstor-sparse-pool-5bvb,default-cstor-testvol-1-678850436-cstor-sparse-pool-tf4k",
      "openebs.io/node-names": "gke-ashish-dev-default-pool-c5bc7514-j4gj,gke-ashish-dev-default-pool-c5bc7514-rhqj,gke-ashish-dev-default-pool-c5bc7514-2j6w"
    },
    "name": "default-cstor-testvol-1-678850436"
  },
  "spec": {
    "capacity": "5G",
    "casType": "cstor",
    "fsType": "ext4",
    "iqn": "iqn.2016-09.com.openebs.cstor:default-cstor-testvol-1-678850436",
    "lun": 0,
    "replicas": "3",
    "targetIP": "10.7.248.146",
    "targetPort": "3260",
    "targetPortal": "10.7.248.146:3260"
  },
  "status": {
    "Message": "",
    "Phase": "",
    "Reason": ""
  }
}
```
-> volume read output for jiva:
```
{
  "apiVersion": "v1alpha1",
  "kind": "CASVolume",
  "metadata": {
    "annotations": {
      "vsm.openebs.io/controller-ips": "10.4.0.9",
      "openebs.io/controller-ips": "10.4.0.9",
      "openebs.io/replica-count": "3",
      "vsm.openebs.io/replica-count": "3",
      "vsm.openebs.io/replica-ips": "10.4.1.8,10.4.0.10,10.4.2.8",
      "openebs.io/replica-status": "running,running,running",
      "vsm.openebs.io/controller-node-name": "gke-ashish-dev-default-pool-c5bc7514-2j6w",
      "openebs.io/replica-ips": "10.4.1.8,10.4.0.10,10.4.2.8",
      "openebs.io/volume-size": "5G",
      "vsm.openebs.io/controller-status": "running,running",
      "vsm.openebs.io/iqn": "iqn.2016-09.com.openebs.jiva:default-jiva-testvol-1-1931265930",
      "vsm.openebs.io/replica-status": "running,running,running",
      "openebs.io/cluster-ips": "10.7.254.120",
      "openebs.io/controller-node-name": "gke-ashish-dev-default-pool-c5bc7514-2j6w",
      "openebs.io/targetportals": "10.7.254.120:3260",
      "vsm.openebs.io/cluster-ips": "10.7.254.120",
      "vsm.openebs.io/targetportals": "10.7.254.120:3260",
      "vsm.openebs.io/volume-size": "5G",
      "openebs.io/controller-status": "running,running",
      "openebs.io/iqn": "iqn.2016-09.com.openebs.jiva:default-jiva-testvol-1-1931265930"
    },
    "name": "default-jiva-testvol-1-1931265930"
  },
  "spec": {
    "capacity": "5G",
    "casType": "jiva",
    "fsType": "ext4",
    "iqn": "iqn.2016-09.com.openebs.jiva:default-jiva-testvol-1-1931265930",
    "lun": 0,
    "replicas": "3",
    "targetIP": "10.7.254.120",
    "targetPort": "3260",
    "targetPortal": "10.7.254.120:3260"
  },
  "status": {
    "Message": "",
    "Phase": "",
    "Reason": ""
  }
}
```

### CODE REVIEW

File: pkg/install/v1alpha1/jiva_volume_0.7.0.go [hunk 1]
```diff
@@ -462,6 +458,7 @@ spec:
         vsm.openebs.io/targetportals: {{ .TaskResult.readlistsvc.clusterIP }}:3260
         openebs.io/controller-ips: {{ .TaskResult.readlistctrl.podIP | default "" | splitList " " | first }}
         openebs.io/cluster-ips: {{ .TaskResult.readlistsvc.clusterIP }}
+        openebs.io/controller-node-name: {{ .TaskResult.readlistctrl.nodeName | default "" }}
```
Comment (kmova) [inline pkg/install/v1alpha1/jiva_volume_0.7.0.go]: call it target-node-name

File: pkg/install/v1alpha1/cstor_volume_0.7.0.go [hunk 2]
```diff
@@ -697,6 +694,7 @@ spec:
         openebs.io/cvr-names: {{ .TaskResult.readlistrep.items | default "" | splitList " " | join "," }}
         openebs.io/node-names: {{ .TaskResult.readlistrep.hostname | default "" | splitList " " | join "," }}
         openebs.io/pool-names: {{ .TaskResult.readlistrep.poolname | default "" | splitList " " | join "," }}
+        openebs.io/controller-node-name: {{ .TaskResult.readlistctrl.podNodeName | default ""}}
```
Comment (kmova) [inline pkg/install/v1alpha1/cstor_volume_0.7.0.go]: call it target-node-name

File: pkg/install/v1alpha1/jiva_volume_0.7.0.go [hunk 3]
```diff
@@ -23,13 +23,7 @@ func JivaVolumeArtifactsFor070() (list ArtifactList) {
...
 }
 
-// jivaVolumeYamlsFor070 returns all the yamls related to jiva volume in a
-// string format
-//
-// NOTE:
-//  This is an implementation of MultiYamlFetcher
-func jivaVolumeYamlsFor070() string {
-	return `
+const jivaRunTask = `
```
Comment (kmova) [inline pkg/install/v1alpha1/jiva_volume_0.7.0.go]: you may want to define the const with the version number. Also, a similar change to run task will be required for 0.6.0 also.. i think.

File: pkg/install/v1alpha1/cstor_volume_0.7.0.go [hunk 4]
```diff
@@ -23,13 +23,7 @@ func CstorVolumeArtifactsFor070() (list ArtifactList) {
...
 }
 
-// cstorVolumeYamlsFor070 returns all the yamls related to cstor volume in a
-// string format
-//
-// NOTE:
-//  This is an implementation of MultiYamlFetcher
-func cstorVolumeYamlsFor070() string {
-	return `
+const cstorRunTask = `
```
Comment (AmitKumarDas) [inline pkg/install/v1alpha1/cstor_volume_0.7.0.go]: Name of constant is not appropriate.
Name of constant should make use of version no. as suffix.
The constant can be a typed string versus a string.

File: pkg/install/v1alpha1/cstor_volume_0.7.0.go [hunk 5]
```diff
@@ -867,4 +865,12 @@ spec:
       name: {{ .Volume.owner }}
 ---
 `
+
+// cstorVolumeYamlsFor070 returns all the yamls related to cstor volume in a
+// string format
+//
+// NOTE:
+//  This is an implementation of MultiYamlFetcher
+func cstorVolumeYamlsFor070() string {
```
Comment (AmitKumarDas) [inline pkg/install/v1alpha1/cstor_volume_0.7.0.go]: We need not have a function if we settle on a constant.

Comment (ashishranjan738) [reply pkg/install/v1alpha1/cstor_volume_0.7.0.go]: yes I agree, for that, I think we should have one more function similar to `ParseArtifactListFromMultipleYamls for parsing string constants.

File: pkg/install/v1alpha1/jiva_volume_0.7.0.go (hunk 3 above)
Comment (ashishranjan738) [reply pkg/install/v1alpha1/jiva_volume_0.7.0.go]: yes I think we should move all templates to constant.

File: pkg/install/v1alpha1/jiva_volume_0.7.0.go (hunk 1 above)
Comment (ashishranjan738) [reply pkg/install/v1alpha1/jiva_volume_0.7.0.go]: done used ```targetNodeName``` instead as facing issue with target-node-name. I think template doesn't support it.

File: pkg/install/v1alpha1/cstor_volume_0.7.0.go (hunk 2 above)
Comment (ashishranjan738) [reply pkg/install/v1alpha1/cstor_volume_0.7.0.go]: done

File: pkg/install/v1alpha1/version.go [hunk 6]
```diff
@@ -27,16 +27,17 @@ const (
...
 
+// Version returns the version in version type if present
 func Version(version string) version {
 	switch version {
 	case "0.7.0":
 		return version070
 	default:
 		return invalidVersion
 	}
-	return invalidVersion
```
Comment (AmitKumarDas) [inline pkg/install/v1alpha1/version.go]: Can you provide corresponding UT for this.

**Pull request merged**